From 1162c9c21497e24e3388689fd2c58e8dfa043bc4 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sun, 13 Jul 2025 12:29:28 +0100
Subject: [PATCH] Update the common submodule

Pick up this commit from the submodule:

  commit b40e534fefb74af32bd496904e44ce9bca1a7b34
  Author: Vadim Rozenfeld <vrozenfe@redhat.com>
  Date:   Thu Jul 10 10:48:04 2025 +1000

    Modify the firstboot script to check the scripts

which is required for commit 594b05d694 ("Modify
configure_pnputil_install script to check pending reboot status and
report PnPUtil execution status") to work properly.

Updates: commit 594b05d6940c8719167d10c0cdfaa253349060ab
(cherry picked from commit 3e08788aaebf96d86452dc2c88d610dc0b72d02e)
---
 common | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Submodule common 0e9caa17..10d2b626:
diff --git a/common/mlcustomize/firstboot.ml b/common/mlcustomize/firstboot.ml
index 6aca4c34..5f2642b0 100644
--- a/common/mlcustomize/firstboot.ml
+++ b/common/mlcustomize/firstboot.ml
@@ -305,13 +305,19 @@ if not exist \"%%scripts_done%%\" (
 :: Pick the next script to run.
 for %%%%f in (\"%%scripts%%\"\\*.bat) do (
   echo running \"%%%%f\"
-  move \"%%%%f\" \"%%scripts_done%%\"
-  pushd \"%%scripts_done%%\"
+  pushd \"%%scripts%%\"
   call \"%%%%~nf\"
   set elvl=!errorlevel!
   echo .... exit code !elvl!
   popd
 
+  if !elvl! NEQ 249 (
+    echo Script succeeded, moving to scripts-done
+    move \"%%%%f\" \"%%scripts_done%%\"
+  ) else (
+    echo Script failed, will retry on next boot
+  )
+
   :: Reboot the computer.  This is necessary to free any locked
   :: files which may prevent later scripts from running.
   shutdown /r /t 0 /y
